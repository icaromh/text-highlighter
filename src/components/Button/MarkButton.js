import React from 'react'
import PropTypes from 'prop-types'

import { MarkIcon } from '../Icons'
import Button from './Button'

const MarkButton = ({ onClick, color }) => (
  <Button
    title="Hightlight a sentence in the document"
    onClick={onClick}
  >
    <MarkIcon color={color} />
  </Button>
)

MarkButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  color: PropTypes.string.isRequired,
}

export default MarkButton
