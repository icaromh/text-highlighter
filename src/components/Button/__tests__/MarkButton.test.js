import React from 'react'
import { shallow } from 'enzyme'

import MarkButton from '../MarkButton'

describe('#MarkButton', () => {
  test('expect to render', () => {
    const wrapper = shallow(<MarkButton color="red" onClick={() => {}} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('expect to perform onClick', () => {
    const onClickMock = jest.fn()
    const wrapper = shallow(<MarkButton color="red" onClick={onClickMock} />)

    wrapper.simulate('click')
    expect(onClickMock).toHaveBeenCalled()
  })
})
