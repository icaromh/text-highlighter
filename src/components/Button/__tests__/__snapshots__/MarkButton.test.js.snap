// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#MarkButton expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MarkButton
    color="red"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <MarkIcon
        color="red"
      />,
      "onClick": [Function],
      "title": "Hightlight a sentence in the document",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": "red",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <MarkIcon
          color="red"
        />,
        "onClick": [Function],
        "title": "Hightlight a sentence in the document",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "red",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
